src/externs.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * @fileoverview This file declares functions that can be declared by external
+ * third party code. Closure compiler advanced optimizations needs this file
+ * to know not to minimize these function names.
+ * @externs
+ */
+
+function shouldPersist(storageInterface, key, value) {}
+function processEvent(storageInterface, helperInterface, name, args) {}
+
+function set(key, value) {}
+function get(key, value) {}
```

Comment:
I'd prefer defining the type and it's methods as externs, rather than a global extern.  This should make the difference between Closure preserving all symbols named processEvent, and just those that are methods of an EventProcessorInterface.

For example, the way Closure defines externs for the Arguments type:
https://github.com/google/closure-compiler/blob/master/externs/es3.js#L230

Reply:
I copied the idea of the code in that file [in a local branch](https://github.com/Alex7Li/measurement-library/blob/externstest/src/externs.js), but it seems that closure still preserves all symbols rather than just the methods. [This file is compiled wrong](https://github.com/Alex7Li/measurement-library/blob/externstest/src/EventProcessor/EventSubclass.js). I'm not sure if I'm misunderstanding your recommendation or if open source closure works differently than you are expecting.

Reply:
This solution will work well enough per offline discussion. Refactored so that we don't declare functions twice.